sr_unix/exttab_parse.c

```diff
@@ -65,31 +67,133 @@ const int parm_space_needed[] =
 	0,
 	0,
 	SIZEOF(void *),
-	SIZEOF(gtm_int_t),
-	SIZEOF(gtm_uint_t),
-	SIZEOF(gtm_long_t),
-	SIZEOF(gtm_ulong_t),
-	SIZEOF(gtm_float_t),
-	SIZEOF(gtm_double_t),
-	SIZEOF(gtm_int_t *) + SIZEOF(gtm_int_t),
-	SIZEOF(gtm_uint_t *) + SIZEOF(gtm_uint_t),
-	SIZEOF(gtm_long_t *) + SIZEOF(gtm_long_t),
-	SIZEOF(gtm_ulong_t *) + SIZEOF(gtm_ulong_t),
-	SIZEOF(gtm_string_t *) + SIZEOF(gtm_string_t),
-	SIZEOF(gtm_float_t *) + SIZEOF(gtm_float_t),
-	SIZEOF(gtm_char_t *),
-	SIZEOF(gtm_char_t **) + SIZEOF(gtm_char_t *),
-	SIZEOF(gtm_double_t *) + SIZEOF(gtm_double_t),
-	SIZEOF(gtm_pointertofunc_t),
-	SIZEOF(gtm_pointertofunc_t *) + SIZEOF(gtm_pointertofunc_t),
-	SIZEOF(gtm_int_t *) + SIZEOF(gtm_int_t),
-	SIZEOF(gtm_int_t *) + SIZEOF(gtm_int_t),
-	SIZEOF(gtm_long_t *) + SIZEOF(gtm_long_t),
-	SIZEOF(gtm_float_t *) + SIZEOF(gtm_float_t),
-	SIZEOF(gtm_double_t *) + SIZEOF(gtm_double_t),
-	SIZEOF(gtm_string_t),  /* gtm_string_t contains a (gtm_long_t) and (gtm_char_t *) */
-	SIZEOF(gtm_string_t),
-	SIZEOF(gtm_string_t)
+	SIZEOF(ydb_int_t),
+	SIZEOF(ydb_uint_t),
+	SIZEOF(ydb_long_t),
+	SIZEOF(ydb_ulong_t),
+	SIZEOF(ydb_float_t),
+	SIZEOF(ydb_double_t),
+	SIZEOF(ydb_int_t *) + SIZEOF(ydb_int_t),
+	SIZEOF(ydb_uint_t *) + SIZEOF(ydb_uint_t),
+	SIZEOF(ydb_long_t *) + SIZEOF(ydb_long_t),
+	SIZEOF(ydb_ulong_t *) + SIZEOF(ydb_ulong_t),
+	SIZEOF(ydb_string_t *) + SIZEOF(ydb_string_t),
+	SIZEOF(ydb_float_t *) + SIZEOF(ydb_float_t),
+	SIZEOF(ydb_char_t *),
+	SIZEOF(ydb_char_t **) + SIZEOF(ydb_char_t *),
+	SIZEOF(ydb_double_t *) + SIZEOF(ydb_double_t),
+	SIZEOF(ydb_pointertofunc_t),
+	SIZEOF(ydb_pointertofunc_t *) + SIZEOF(ydb_pointertofunc_t),
+	SIZEOF(ydb_int_t *) + SIZEOF(ydb_int_t),
+	SIZEOF(ydb_int_t *) + SIZEOF(ydb_int_t),
+	SIZEOF(ydb_long_t *) + SIZEOF(ydb_long_t),
+	SIZEOF(ydb_float_t *) + SIZEOF(ydb_float_t),
+	SIZEOF(ydb_double_t *) + SIZEOF(ydb_double_t),
+	SIZEOF(ydb_string_t),  /* ydb_string_t contains a (ydb_long_t) and (ydb_char_t *) */
+	SIZEOF(ydb_string_t),
+	SIZEOF(ydb_string_t)
+};
+
+/* This table is searched serially so the search priority is:
```

Comment:
It only matters when a shared library is first attached and the definition file is parsed and the types looked up. All calls after the first have no penalty. So upside for any improvement is small, especially with most definition files only having a small handful of calls/types in them.

Sorting and making a binary search would be a nice change but the cost/benefit ratio is a bit low so I could see creating an issue as a nice to have but it might get fairly old before it gets handled. I don't think it is important enough to worry about in this commit though.

I'm also ok with types that aren't ydb_* having a slight penalty. :-D